File: android/app/src/main/java/app/tourism/data/dto/place/CoordinatesDto.kt
package app.tourism.data.dto.place

import app.tourism.data.dto.PlaceLocation

data class CoordinatesDto(
    val latitude: String?,
    val longitude: String?
) {
    fun toPlaceLocation(name: String): PlaceLocation? {
        try {
            return PlaceLocation(
                name,
                latitude!!.toDouble(),
                longitude!!.toDouble()
            )
        } catch (e: Exception) {
            e.printStackTrace()
            return null
        }
    }
}

File: iphone/Maps/Tourism/Data/Network/DTO/Details/CoordinatesDTO.swift
import Foundation

struct CoordinatesDTO: Codable {
  let latitude: String?
  let longitude: String?
  
  func toPlaceLocation(name: String) -> PlaceLocation? {
    guard let latitude = latitude, let longitude = longitude,
          let lat = Double(latitude), let lon = Double(longitude) else {
      return nil
    }
    return PlaceLocation(name: name, lat: lat, lon: lon)
  }
}
